tangos/input_handlers/ramsesHOP.py

```diff
@@ -0,0 +1,224 @@
+
+        bridge = pynbody.bridge.OrderBridge(f1, f2, monotonic=False)
+
+        return bridge.fuzzy_match_catalog(halo_min, halo_max, threshold=threshold, only_family=pynbody.family.dm,
+                                          groups_1=h1, groups_2=h2)
+
+    def _is_able_to_load(self, ts_extension):
+        import pynbody
+        filepath = self._extension_to_filename(ts_extension)
+        try:
+            f = pynbody.load(filepath)
+            if self.quicker:
+                logger.warn("Pynbody was able to load %r, but because 'quicker' flag is set we won't check whether it can also load the halo files", filepath)
+            else:
+                h = pynbody.halo.hop.HOPCatalogue(f)
+            return True
+        except (IOError, RuntimeError):
+            return False
+
+
+
+class RamsesAdaptaHOPInputHandler(PynbodyInputHandler):
+    """ Handling Ramses outputs with AdaptaHOP halo and subhalo finding """
+
+    patterns = ["output_0????"]
+    auxiliary_file_patterns = ["tree_bricks???"]
+
+    def match_objects(self, ts1, ts2, halo_min, halo_max, dm_only=False, threshold=0.005,
+                      object_typetag="halo", output_handler_for_ts2=None):
+        import pynbody
+        f1 = self.load_timestep(ts1).dm
+        h1 = self._construct_halo_cat(ts1, object_typetag)
+
+        if output_handler_for_ts2 is None:
+            f2 = self.load_timestep(ts2).dm
+            h2 = self._construct_halo_cat(ts2, object_typetag)
+        else:
+            f2 = output_handler_for_ts2.load_timestep(ts2).dm
+            h2 = output_handler_for_ts2._construct_halo_cat(ts2, object_typetag)
+
+        bridge = pynbody.bridge.OrderBridge(f1, f2, monotonic=False)
+
+        return bridge.fuzzy_match_catalog(halo_min, halo_max, threshold=threshold, only_family=pynbody.family.dm,
+                                          groups_1=h1, groups_2=h2)
+
+    def _is_able_to_load(self, ts_extension):
+        import pynbody
+        filepath = self._extension_to_filename(ts_extension)
+        try:
+            f = pynbody.load(filepath)
+            if self.quicker:
+                logger.warn("Pynbody was able to load %r, but because 'quicker' flag is set we won't check whether it can also load the halo files", filepath)
+            else:
+                h = f.halos(index_parent=False)
+                return isinstance(h, pynbody.halo.adaptahop.BaseAdaptaHOPCatalogue)
+            return True
+        except (IOError, RuntimeError):
+            return False
+
+    def _exclude_adaptahop_precalculated_properties(self):
+        return ["members", "timestep", "level", "host_id", "first_subhalo_id", "next_subhalo_id",
+                "pos_x", "pos_y", "pos_z", "pos", "vel_x", "vel_y", "vel_z",
+                "angular_momentum_x", "angular_momentum_y", "angular_momentum_z",
+                "contaminated", "m_contam", "mtot_contam", "n_contam", "ntot_contam"]
+
+    def _include_additional_properties_derived_from_adaptahop(self):
```

Comment:
Misleading method name as above. By the way, why make these methods at all? You could just define them as static data within the class.